### PR TITLE
Updated timeseries_dataset.py

### LINKED ISSUE #17410: Issues in tf.keras.utils.timeseries_dataset_from_array
Issue 1:
The line 111 in timeseries_dataset_from_array [https://github.com/keras-team/keras/blob/v2.11.0/keras/utils/timeseries_dataset.py#L24-L266] must be corrected
from     input_data = data[:-10]
to input_data = data[:-1].
The input data can be used all except the last one to build dataset for predicted one time step forward.
 In general, for predicting delay time steps ahead with sequence_length, the code must be used with
    tf.keras.utils.timeseries_dataset_from_array(
        data = data[:-delay],
        targets = target[sequence_length+delay-1:]
        sequence_length,
        sequence_stride=1,
        sampling_rate=1,
        batch_size=128,
        shuffle=False,
        seed=None,
        start_index=None,
        end_index=None
    )
Issue 2: shuffle must be modified so that the shuffle occurs in batch level, not changing the chlonogical order of the sequence. To avoid this problem instead of shuffle = True, we must use the method, train_dataset.shuffle(10) for example.

### PR DESCRIPTION
**Pull request by sushreebarsa**
Fixes #17410

### CODE REVIEW

**Comment by fchollet** [comment]
The current example code snippet runs as-is without issue. Please explain the nature of the change.

**Comment by sushreebarsa** [comment]
@fchollet You were right. The example code is running fine. Closing this PR.
Thank you for looking into this.
